src/config.py

```diff
@@ -2,85 +2,119 @@
-    # noinspection PyTypeChecker
-    return _cast_str(env_var, cast_to)
+    return env_var
 
 
-def _cast_str(str_to_cast: str, cast_to: Type[T]) -> T:
+def _get_bool_env_variable(var_name: str, default: Optional[bool] = None) -> bool:
     """
-    Take a string and cast it to necessary primitive builtin types. Tested with int, float, and bool.
-    For bool, this detects if the value is in the case-insensitive sets {"True", "T", "1"} or {"False", "F", "0"}
-    and raises a ValueError if not. For example _cast_str("False", bool) -> False
+    Read an environment variable and attempts to cast to bool, with settings to allow defaults.
+    For bool casting we have the problem where bool("False") == True
+    but this function fixes that so get_bool_env_variable("False") == False
+
+    For public use please use EnvVariable.
 
     Parameters
     ----------
-    str_to_cast : str
-        The string that is going to be casted to the type
-    cast_to : Type[T]
-        The type to cast to e.g. bool
+    var_name : str
+        The name of the environment variable to retrieve.
+    default : Optional[bool] = None
+        Default return value if the environment variable does not exist.
 
     Returns
     -------
-    The string casted to type T defined by cast_to.
+    bool
+        The environment variable, or default if it does not exist
 
     Raises
     ------
-    ValueError if [cast_to] is not compatible with the value stored.
+    ValueError
+        If allow_empty is False and the environment variable is empty string or None
     """
-    # Special cases i.e. casts that aren't of the form int("7") -> 7
-    if cast_to == bool:
-        # For bool we have the problem where bool("False") == True but we want this function to return False
-        truth_values = {"true", "t", "1"}
-        false_values = {"false", "f", "0", ""}
-        if str_to_cast is None or str_to_cast.lower() in false_values:
-            return False
-        elif str_to_cast.lower() in truth_values:
-            return True
-        raise ValueError(f"{str_to_cast} being casted to bool but is not in {truth_values} or {false_values}")
-    # General case
-    return cast_to(str_to_cast)
+    env_variable = _get_env_variable(var_name, str(default))
+    truth_values = {"true", "t", "1"}
+    false_values = {"false", "f", "0"}
+    if env_variable.lower() in truth_values:
+        return True
+    elif env_variable.lower() in false_values:
+        return False
+    raise ValueError(f"Environment variable {var_name}={env_variable} being casted to bool "
+                     f"but is not in {truth_values} or {false_values}")
+
+
+class EnvVariable:  # pylint: disable=too-few-public-methods
+    """Encapsulates all environment variable fetching, ensuring proper defaults and types."""
+
+    STATSNZ_API_KEY = _get_env_variable("STATSNZ_API_KEY")
+    LINZ_API_KEY = _get_env_variable("LINZ_API_KEY")
+    LRIS_API_KEY = _get_env_variable("LRIS_API_KEY")
```

Comment:
Please consider removing this line or setting `allow_empty` to True. This will prevent the "KeyError: 'Environment variable LRIS_API_KEY not set, and allow_empty is False'" error. If you decide to remove it, you should also remove the LRIS_API_KEY entry from `api_keys.env` since it is not being used.
```suggestion
    LRIS_API_KEY = _get_env_variable("LRIS_API_KEY", allow_empty=True)
```

Reply:
I'll remove it. I'll also remove LRIS from the vectory fetching code.